### ISSUE TITLE
Problem with initialValues and redux-form/immutable

### ISSUE THREAD

**Issue by iBoB-Code** [erikras/redux-form#4246]
Hello guys,

### The problem
https://codesandbox.io/s/3o4j9y375

### What I have
Ok, this one is really tricky... I have the following code : 
`
Form = connect((state, props) => ({
  formValues: getFormValues('myForm')(state),
  initialValues: props.values,
}))(Form);
`

### What I expect
Having in my Form props both formValues and initialValues.

### Where did it go wrong?
First I just wanted formValues to be in my Form props : I encountered an issue with that because i am using **IMMUTABLE** so I had to import like this
`import { getFormValues } from 'redux-form/immutable';`
Instead of like this
`import { getFormValues } from 'redux-form';`
Once that fixed I wanted to add some initialValues to my form as showed above and THAT is is the problem.
I console logged my props in Form and initialValues are passed but not in the JSON format that I used but in an **IMMUTABLE MAP**...
But in this format, the values are not injected in the form.

### TL/DR
import redux-form/immutable -> can use **getFormValues** but **initialValues** are stored as immutable Map so not working
import redux-form-> cannot use **getFormValues** but **initialValues** are stored in JSON so it's working

**Comment by gustavohenke** [comment]
Although I have been around for a while, I never played much with redux-form using ImmutableJS.
My guess is that it is correct to expect the values to be `Immutable.Map`, if that's what you are storing...

**Comment by lynkx** [comment]
@iBoB-Code I think the reason you're getting this behavior is that your form reducer (the one in the store.js) is being imported from redux-form (instead of redux-form/immutable).

Check the sandbox here (with proper imports):
https://codesandbox.io/s/r7loq51xpo

Now both submit buttons will produce the same output.
